### PR TITLE
New design scroll editor

### PR DESCRIPTION
**Pull request by zmbq**
Bronson,

This pull request contains a redesign of the scroll editor, so that it looks similar to the rest of the system. It also contains a few other minor UI fixes.

### CODE REVIEW

File: src/components/misc/CorruptedStateDialog.vue [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+         ref="corruptedModalRef"
+        header-class="title-header"
+        footer-class="title-footer"
+    >
+        <div>
+            <label> We are facing some issues, please reload this page</label>
+        </div>
+        <template v-slot:modal-footer>
+            <div class="w-100">
+                <b-button variant="outline-primary" @click="reloadPage()"
```
Comment (Bronson-Brown-deVost) [inline src/components/misc/CorruptedStateDialog.vue:13]: Very nice!

File: src/components/text/text-sign.vue [hunk 2]
```diff
@@ -1,41 +1,75 @@
...
+                ref="signMenu"
+                @blur="closeSignMenu($event)"
+            >
+                <ul>
+                    <li>
+                        <p @click="openEditSignModal()">Edit sign</p>
+                    </li>
+                    <li>
+                        <p @click="deleteSignInterpretation(si)"
+                            >Delete sign</p
```
Comment (Bronson-Brown-deVost) [inline src/components/text/text-sign.vue:34]: We probably should pipe these titles on though i18n.

File: src/i18n/en/misc.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 export default {
-    cancel: 'Cancel',
+    cancel: ' Delete ROI',
```
Comment (Bronson-Brown-deVost) [inline src/i18n/en/misc.ts:2]: Will it be a problem that these is a space before the "D" here?

File: src/models/text.ts [hunk 4]
```diff
@@ -147,6 +147,12 @@ class Sign {
 }
 
 class SignInterpretation {
+    private static nextId = 0;
+    public static get nextAvailableId() {
+        SignInterpretation.nextId --;
```
Comment (Bronson-Brown-deVost) [inline src/models/text.ts:152]: I don't have the time to read through the code here, but it looks like you decrement the nextId every time someone reads nextAvailableId, was that intended.  I don't love a getter that has a mutation side effect, or am I simply reading the code wrong?  My apologies if so.

File: src/models/text.ts [hunk 5]
```diff
@@ -187,12 +193,51 @@ class SignInterpretation {
         }
     }
 
+    public findAttributeIndex(attributeValueId: number) {
+        return this.attributes.findIndex(attr => attr.attributeValueId === attributeValueId);
+    }
+
     public get isReconstructed(): boolean {
         return !!this.attributes.find(attr => attr.attributeString === 'is_reconstructed');
```
Comment (Bronson-Brown-deVost) [inline src/models/text.ts:201]: I know this is old code, and !! is cute, but I tend to prefer "!== undefined" to Javascript coercion.  A guy can dream that someday they will clean up all the coercion/equality junk in Javascript ... ok, I know it won't really happen.

**Review by Bronson-Brown-deVost — APPROVED** [review]
Looks good to me.  Many thanks!

File: src/components/misc/CorruptedStateDialog.vue (hunk 1 above)
Comment (zmbq) [reply src/components/misc/CorruptedStateDialog.vue:13]: Yes, we should start emitting this event soon.

File: src/models/text.ts (hunk 4 above)
Comment (zmbq) [reply src/models/text.ts:152]: It is intended, yes. This is not a regular property, and nobody binds to it - this is the property that provides the next frontend-only ID for new objects. If someone calls it too much, we'll just get holes - that's fine, we have about 2^31 negative numbers (maybe 2^63), and this is reset whenever you refresh the page.

**Pull request merged**